components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -2143,44 +2195,53 @@ void JsonRpcService::OnDiscoverAssetsCompleted(
   }
 }
 
-void JsonRpcService::DiscoverAssetsInternal(
+void JsonRpcService::DiscoverAssets(
     const std::string& chain_id,
     mojom::CoinType coin,
     const std::vector<std::string>& account_addresses,
+    bool update_next_asset_discovery_from_block,
+    const std::string from_block,
+    const std::string to_block,
     DiscoverAssetsCallback callback) {
-  if (coin != mojom::CoinType::ETH || chain_id != mojom::kMainnetChainId) {
-    std::move(callback).Run(
-        std::vector<mojom::BlockchainTokenPtr>(),
+  // Asset discovery only supported on select EVM chains
+  if (coin != mojom::CoinType::ETH ||
+      !base::Contains(GetAssetDiscoverySupportedChains(), chain_id)) {
+    JsonRpcService::CompleteDiscoverAssets(
```

Comment:
Just CompleteDiscoverAssets should be enough, same as other places.

Reply:
There are also a multiple JsonRpcService::DiscoverAssets instances in this file which should just be DiscoverAssets.